gitlab/utils.py

```diff
@@ -161,3 +161,24 @@ def warn(
         stacklevel=stacklevel,
         source=source,
     )
+
+
+    def _validate_attrs(
+            data: Dict[str, Any],
+            attributes: RequiredOptional,
+            excludes: Optional[list] = None
+        ) -> None:
+        if excludes is None:
+            excludes = []
+
+        required = [k for k in attributes.required if k not in excludes]
+        missing = [ attr for attr in required if attr not in data ]
```

Comment:
Our formatter/linter will probably complain about this style. Running `tox -e black` should fix it. You can find more details on the linters we run here:

https://github.com/python-gitlab/python-gitlab/blob/main/CONTRIBUTING.rst

Reply:
done.